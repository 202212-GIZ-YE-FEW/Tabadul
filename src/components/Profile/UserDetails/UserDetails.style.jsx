import styled from "@emotion/styled";
import Image from "next/image";

export const MainContainer = styled.div`
    display: flex;
    justify-content: center;
    align-items: center;
    flex-direction: row;
    gap: 102px;
    @media screen and (max-width: 1400px) {
        flex-direction: column;
        gap: 60px;
    }
`;

export const ImageSection = styled.div`
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
`;

export const ProfileImg = styled(Image)`
    max-width: 100%;
    margin-bottom: 21px;
`;

export const Text = styled.span`
    font-style: normal;
    font-weight: 400;
    font-size: 24px;
    text-align: center;
    margin: 8px;
`;

export const UserInfo = styled.div`
    display: flex;
    position: relative;
    justify-content: space-between;
    background: #f5f5f5;
    border-radius: 20px;
    height: 161px;
    width: 1242px;
    max-width: 100%;
    /* gap: 10px; */
    @media screen and (max-width: 991px) {
        display: flex;
        justify-content: space-between;
        flex-direction: column;
        height: 100%;
        width: 100%;
        padding: 27px 0 31px;
    }
    @media screen and (max-width: 768px) {
        padding: 38px 0 0px 31px;
    }
`;

export const Button = styled.button`
    background: #e8e8e8;
    border-radius: 0px 20px 20px 0px;
    border: none;
    width: 81px;

    @media screen and (max-width: 768px) {
        background: #f5f5f5;
        position: absolute;
        top: 15px;
        right: -2px;
    }
`;
export const PenImage = styled(Image)`
    @media screen and (max-width: 768px) {
        height: 29px;
        width: 28px;
    }
`;
export const Section = styled.div`
    display: flex;
    flex-direction: column;
    align-items: start;
    width: 364px;
    height: 68px;
    &:not(:last-child) {
        padding: 20px 10px 20px 10px;
    }
    @media screen and (max-width: 991px) {
        justify-content: space-between;
        align-items: left;
        height: 100%;
        width: 100%;
    }
    @media screen and (max-width: 768px) {
        background: #f5f5f5;
        &:not(:last-child) {
            padding: 0;
            padding-bottom: 23px;
        }
        & > span {
            font-size: 20px;
            font-weight: 400;
            line-height: 20px;
            &:first-of-type {
                padding-bottom: 23px;
            }
        }
    }
    @media screen and (max-width: 320px) {
        & > span {
            font-size: 18px;
        }
    }
`;
