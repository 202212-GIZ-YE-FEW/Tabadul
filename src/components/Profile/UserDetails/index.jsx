import { collection, getDocs } from "firebase/firestore";
import Link from "next/link";
import React, { useEffect, useState } from "react";

import { auth, db } from "@/utils/firebase";

import {
    Button,
    ImageSection,
    MainContainer,
    PenImage,
    ProfileImg,
    Section,
    Text,
    UserInfo,
} from "./UserDetails.style";
import Pen from "../../../../public/images/pen.svg";
import ProfileImage from "../../../../public/images/ProfilePlaceholder.svg";
import { useRouter } from "next/router";
import { useTranslation } from "next-i18next";

const UserDetails = () => {
    const localuser = {
        language: "En",
    };
    const [currentuser, setCurrentUser] = useState();
    const usersCollRef = collection(db, "users");
    const router = useRouter();
    const { t } = useTranslation("profile");
    const translateMe = (word) => {
        return t(word);
    };
    useEffect(() => {
        const getalluser = async () => {
            try {
                const usersData = await getDocs(usersCollRef);
                const usersDocs = usersData.docs
                    .map((loc) => {
                        return { ...loc.data(), id: loc.id };
                    })
                    .filter((ele) => {
                        return ele.uid === auth.currentUser.uid;
                    });
                setCurrentUser(usersDocs[0]);
            } catch (err) {
                console.error(err);
            }
        };
        getalluser();
    }, []);

    return (
        <>
            <MainContainer>
                <ImageSection>
                    <ProfileImg
                        src={ProfileImage}
                        alt='ProfileImage'
                        width={222}
                        height={222}
                        priority
                    />
                    <Text>
                        {currentuser?.name} {currentuser?.lname}
                    </Text>
                    <Text>{currentuser?.location}</Text>
                </ImageSection>
                <UserInfo>
                    <Section>
                        <Text>
                            {translateMe("name")} : {currentuser?.name}{" "}
                            {currentuser?.lname}
                        </Text>
                        <Text>{currentuser?.location}</Text>
                    </Section>
                    <Section>
                        <Text>
                            {translateMe("phone")} : {currentuser?.phone}
                        </Text>
                        <Text>
                            {translateMe("email")} :{" "}
                            {currentuser?.email.substring(0, 16)}
                            ..
                        </Text>
                    </Section>
                    <Section>
                        <Text>
                            {translateMe("language")} : {localuser.language}
                        </Text>
                    </Section>
                    <Button router={router.locale}>
                        <Link href='/Updateuser'>
                            <PenImage src={Pen} alt='Edit' />
                        </Link>
                    </Button>
                </UserInfo>
            </MainContainer>
        </>
    );
};

export default UserDetails;
