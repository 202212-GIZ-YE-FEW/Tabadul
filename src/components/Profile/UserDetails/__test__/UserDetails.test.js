import renderer from "react-test-renderer";

import UserDetails from "..";

it("renders correctly", () => {
    const tree = renderer.create(<UserDetails />).toJSON();
    expect(tree).toMatchSnapshot();
});
