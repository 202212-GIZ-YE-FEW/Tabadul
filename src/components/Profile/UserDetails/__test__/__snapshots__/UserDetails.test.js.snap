// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="css-il3fo8"
>
  <div
    className="css-omk73n"
  >
    <img
      alt="ProfileImage"
      className="css-vn9ci3"
      data-nimg="1"
      decoding="async"
      height={222}
      onError={[Function]}
      onLoad={[Function]}
      src="/_next/image?url=%2Fimg.jpg&w=640&q=75"
      srcSet="/_next/image?url=%2Fimg.jpg&w=256&q=75 1x, /_next/image?url=%2Fimg.jpg&w=640&q=75 2x"
      style={
        Object {
          "color": "transparent",
        }
      }
      width={222}
    />
    <span
      className="css-35o1u0"
    >
       
    </span>
    <span
      className="css-35o1u0"
    />
  </div>
  <div
    className="css-1e4byr1"
  >
    <div
      className="css-1yw0kxu"
    >
      <span
        className="css-35o1u0"
      >
        name
         :  
         
      </span>
      <span
        className="css-35o1u0"
      />
    </div>
    <div
      className="css-1yw0kxu"
    >
      <span
        className="css-35o1u0"
      >
        phone
         :  
      </span>
      <span
        className="css-35o1u0"
      >
        email
         :  
        ..
      </span>
    </div>
    <div
      className="css-1yw0kxu"
    >
      <span
        className="css-35o1u0"
      >
        language
         : 
        En
      </span>
    </div>
    <button
      className="css-jcwlzp"
    >
      <a
        href="/Updateuser"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onTouchStart={[Function]}
      >
        <img
          alt="Edit"
          className="css-1oygifg"
          data-nimg="1"
          decoding="async"
          height={40}
          loading="lazy"
          onError={[Function]}
          onLoad={[Function]}
          src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
          srcSet="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
          style={
            Object {
              "color": "transparent",
            }
          }
          width={40}
        />
      </a>
    </button>
  </div>
</div>
`;
